# Question 1

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


Minitab File : Cutlets.mtw

In [1]:
import numpy as np
import pandas as pd
import scipy
from scipy import stats
from scipy.stats import norm

In [2]:
Cutlets=pd.read_csv("C:\\Users\\avina\\Documents\\Data Science\\Assignments\\003. Hypothesis testing\\Cutlets.csv")
Cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

Let,

H0 : There is no significant difference in diameter of the cutlet

H1 : There is significant difference in diameter of the cutlet

Level of significance = 5%

In [3]:
UnitA=pd.Series(Cutlets.iloc[:,0])
UnitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [4]:
UnitB=pd.Series(Cutlets.iloc[:,1])
UnitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [5]:
P_value=stats.ttest_ind(UnitA,UnitB)
P_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [34]:
p_value=P_value[1]
print(p_value)

0.4722394724599501


In [35]:
alpha=0.05
if p_value<=alpha:
    print("We reject H0, ",'\n',"There is significant difference in diameter of the cutlet")
else:
    print("We accept H0, ",'\n',"There is no significant difference in diameter of the cutlet")

We accept H0,  
 There is no significant difference in diameter of the cutlet


#  Question 2

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [7]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [8]:
TAT=pd.read_csv("C:\\Users\\avina\\Documents\\Data Science\\Assignments\\003. Hypothesis testing\\LabTAT.csv")
TAT

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

let,

H0 : There is no significant diference in average TAT

H1 : There is a significant difference in average TAT

Level of significance = 5%

alpha=0.05

In [9]:
p_value=stats.f_oneway(TAT['Laboratory 1'],TAT['Laboratory 2'],TAT['Laboratory 3'],TAT['Laboratory 4'])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [10]:
pvalue=p_value[1]
pvalue

2.1156708949992414e-57

In [36]:
alpha=0.05
if pvalue <= alpha:
    print('We reject Null hypothesis H0, and accept Alternate hypothesis H1')
else:
    print('We accept Null hypothesis H0, and reject Alternate hypothesis H1')

We reject Null hypothesis H0, and accept Alternate hypothesis H1


Here,

We reject Null hypothesis H0, and accept Alternate hypothesis H1

ie, 

There is significant difference in average TAT of sample

# Question 3

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


In [12]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [13]:
Buyer=pd.read_csv('C:\\Users\\avina\\Documents\\Data Science\\Assignments\\003. Hypothesis testing\\BuyerRatio.csv')
Buyer

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Let, 
Null hypothesis be,

H0 : All proportions are equal

and Alternate hypothesis be,

H1 : Not all proportions are equal

Level of significance = 5%

Alpha = 0.05

In [14]:
from scipy.stats import chi2_contingency

In [15]:
table=[[50,142,131,70],[435,1523,1356,750]]
table

[[50, 142, 131, 70], [435, 1523, 1356, 750]]

In [16]:
exp=stats.chi2_contingency(table)
exp

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [17]:
# Perform the chi-square test
chi2, p, dof, expected = chi2_contingency(table)

print("Chi-square statistic:", chi2)
print("p-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies:\n", expected)

Chi-square statistic: 1.595945538661058
p-value: 0.6603094907091882
Degrees of freedom: 3
Expected frequencies:
 [[  42.76531299  146.81287862  131.11756787   72.30424052]
 [ 442.23468701 1518.18712138 1355.88243213  747.69575948]]


In [18]:
alpha=0.05
if p<=alpha:
    print("reject null hypothesis")
else:
    print("accept null hypothesis")

accept null hypothesis


Here,

Chi-square statistic: 1.595945538661058
p-value: 0.6603094907091882
Degrees of freedom: 3
alpha = 0.05

since the p-value is greater than alpha 

we accept null hypothesis and accept alternate hypothesis

ie,

male-female buyer rations are  similar across regions

# Question 4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [19]:
data=pd.read_csv("C:\\Users\\avina\\Documents\\Data Science\\Assignments\\003. Hypothesis testing\\Costomer+OrderForm.csv")
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

Here let the null hypothesis be,

H0: The defective % doesnot varies by centre

and the alternate hypothesis be,

H1: The defective %  varies by centre

Level of significance = 5%

alpha=0.05

In [20]:
data.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [21]:
print(data['Phillippines'].value_counts(),'\n',
     data['Indonesia'].value_counts(),'\n',
     data['Malta'].value_counts(),'\n',
     data['India'].value_counts(),'\n')

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 
 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 
 Error Free    269
Defective      31
Name: Malta, dtype: int64 
 Error Free    280
Defective      20
Name: India, dtype: int64 



In [22]:
contigency_table=[[271,267,269,280],[29,33,31,20]]
print("contigency_table: ",contigency_table)

contigency_table:  [[271, 267, 269, 280], [29, 33, 31, 20]]


In [23]:
# Calculating expected values

stat,p,df,exp=stats.chi2_contingency(contigency_table)
print("Statistics: ",stat,'\n',
     "p_value: ",p,'\n',
     "Degree_of_freedom: ",df,'\n',
     "Expected_values: ",exp)

Statistics:  3.858960685820355 
 p_value:  0.2771020991233135 
 Degree_of_freedom:  3 
 Expected_values:  [[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


In [24]:
observed_values=np.array([271,267,269,280,29,33,31,20])
expected_values=np.array([271.75,271.75,271.75,271.75,28.25,28.25,28.25,28.25])

In [25]:
test_statistic,p_value=stats.chisquare(observed_values,expected_values,ddof=df)
print("Test Statistics: ",test_statistic,'\n',
     "P_value: ",p_value)

Test Statistics:  3.858960685820355 
 P_value:  0.4254298144535761


In [30]:
alpha=0.05
p=0.4254298144535761
if p<=alpha:
    print("We reject null hypothesis H0, ",'\n',"The defective % doesnot varies by centre")
else:
    print("We accept null hypothesis H0, ",'\n',"The defective % varies by centre")

We accept null hypothesis H0,  
 The defective % varies by centre
